### PR TITLE
Changing learning rate from type Input(float) to Input(tensor)

### PR DESCRIPTION
**Pull request by abhinavarora**
Fixes https://github.com/PaddlePaddle/Paddle/issues/4577

### CODE REVIEW

File: paddle/operators/sgd_op.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ class SGDOpKernel : public framework::OpKernel<T> {
     auto param = ctx.Input<Tensor>("param");
     auto grad = ctx.Input<Tensor>("grad");
     auto param_out = ctx.Output<Tensor>("param_out");
-    float lr = *ctx.Input<float>("learning_rate");
+    float lr = ctx.Input<Tensor>("learning_rate")->data<float>()[0];
```
Comment (jacquesqiao) [inline paddle/operators/sgd_op.h]: here should also use Tensor and change it to Eigen data structure for GPU computing~
